File: app/src/main/java/com/example/dev/hazikura/fragment/Remainder/CalAdapter.java
package com.example.dev.hazikura.fragment.Remainder;

import android.content.Context;
import android.database.Cursor;
import android.database.CursorIndexOutOfBoundsException;
import android.graphics.Color;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AbsListView;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.example.dev.hazikura.R;
import com.example.dev.hazikura.fragment.Household.DBAdapter;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Locale;

/**
 * Created by yokoro
 **/

public class CalAdapter extends BaseAdapter {
    DBAdapter dbAdapter;

    public static List<Date> dateArray = new ArrayList();
    private Context mContext;
    private DateManager mDateManager;
    private LayoutInflater mLayoutInflater;

    //カスタムセルを拡張したらここでWigetを定義
    private static class ViewHolder {
        public TextView dateText;
    }

    public CalAdapter(Context context){
        mContext = context;
        mLayoutInflater = LayoutInflater.from(mContext);
        mDateManager = new DateManager();
        dateArray = mDateManager.getDays();
    }

    @Override
    public int getCount() {
        return dateArray.size();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View v = convertView;
        ViewHolder holder;
        if (convertView == null) {
            convertView = mLayoutInflater.inflate(R.layout.calendar_cell, null);
            holder = new ViewHolder();
            holder.dateText = convertView.findViewById(R.id.dateText);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder)convertView.getTag();
        }

        //セルのサイズを指定
        float dp = (mContext.getResources().getDisplayMetrics().density);
        AbsListView.LayoutParams params = new AbsListView.LayoutParams(parent.getWidth()/7 - (int)dp, (parent.getHeight() - (int)dp * mDateManager.getWeeks() ) / mDateManager.getWeeks());
        convertView.setLayoutParams(params);

        //日付のみ表示させる
        SimpleDateFormat dateFormat = new SimpleDateFormat("d", Locale.US);
        holder.dateText.setText(dateFormat.format(dateArray.get(position)));

        SimpleDateFormat format = new SimpleDateFormat("yyyy/MM/dd", Locale.JAPAN);

        //当月以外のセルをグレーアウト
        if(!searchPlan(format.format(dateArray.get(position))).equals("")){
            convertView.setBackgroundColor(Color.rgb(210,240,200));
        }
        else
            if (mDateManager.isCurrentMonth(dateArray.get(position))){
            convertView.setBackgroundColor(Color.WHITE);
            }else {
            convertView.setBackgroundColor(Color.LTGRAY);
        }

        //日曜日を赤、土曜日を青に
        int colorId;
        switch (mDateManager.getDayOfWeek(dateArray.get(position))){
            case 1:
                colorId = Color.RED;
                break;
            case 7:
                colorId = Color.BLUE;
                break;

            default:
                colorId = Color.BLACK;
                break;
        }
        holder.dateText.setTextColor(colorId);

        return convertView;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    //表示月を取得
    public String getTitle(){
        SimpleDateFormat format = new SimpleDateFormat("yyyy.MM", Locale.JAPAN);
        return format.format(mDateManager.mCalendar.getTime());
    }

    public String getDate(){
        SimpleDateFormat format = new SimpleDateFormat("yyyy/MM/dd", Locale.JAPAN);
        return format.format(mDateManager.mCalendar.getTime());
    }

    //翌月表示
    public void nextMonth(){
        mDateManager.nextMonth();
        dateArray = mDateManager.getDays();
        this.notifyDataSetChanged();
    }

    //前月表示
    public void prevMonth(){
        mDateManager.prevMonth();
        dateArray = mDateManager.getDays();
        this.notifyDataSetChanged();
    }

    public String searchPlan(String date) {
        String plan;

        dbAdapter = new DBAdapter(this.mContext);
        dbAdapter.readDB();                         // DBの読み込み(読み込みの方)

        String column = "date";          //検索対象のカラム名
        String[] name = {date};            //検索対象の文字

        // DBの検索データを取得 入力した文字列を参照してDBの品名から検索
        Cursor c = dbAdapter.searchDB("remainder",null, column, name);

        c.moveToFirst();
        try{
            plan = c.getString(2);
            Log.d(date,plan);
        }
        catch (CursorIndexOutOfBoundsException e){

            Log.d(date,"null");
            plan = "";
        }
        c.close();
        dbAdapter.closeDB();        // DBを閉じる

        return plan;
    }

}

File: app/src/main/java/com/example/dev/hazikura/fragment/remainder/CalAdapter.kt
package com.example.dev.hazikura.fragment.remainder

import android.database.CursorIndexOutOfBoundsException
import android.graphics.Color
import android.support.v4.app.FragmentActivity
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AbsListView
import android.widget.BaseAdapter
import android.widget.TextView

import com.example.dev.hazikura.R
import com.example.dev.hazikura.fragment.household.DBAdapter

import java.text.SimpleDateFormat
import java.util.ArrayList
import java.util.Date
import java.util.Locale

/**
 * Created by yokoro
 */

class CalAdapter(private val mContext: FragmentActivity) : BaseAdapter() {
    private lateinit var dbAdapter: DBAdapter
    private val mDateManager: DateManager = DateManager()
    private val mLayoutInflater: LayoutInflater = LayoutInflater.from(mContext)

    //表示月を取得
    val title: String
        get() {
            val format = SimpleDateFormat("yyyy.MM", Locale.JAPAN)
            return format.format(mDateManager.mCalendar.time)
        }

    val date: String
        get() {
            val format = SimpleDateFormat("yyyy/MM/dd", Locale.JAPAN)
            return format.format(mDateManager.mCalendar.time)
        }

    //カスタムセルを拡張したらここでWigetを定義
    private class ViewHolder {
        var dateText: TextView? = null
    }

    init {
        dateArray = mDateManager.days
    }

    override fun getCount(): Int {
        return dateArray.size
    }

    override fun getView(position: Int, view: View?, parent: ViewGroup): View {
        var convertView = view
        var holder = ViewHolder()
        if (convertView == null) {
            convertView = mLayoutInflater.inflate(R.layout.calendar_cell, null)
            holder.dateText = convertView!!.findViewById(R.id.dateText)
            convertView.tag = holder
        } else {
            holder = convertView.tag as ViewHolder
        }

        //セルのサイズを指定
        val dp = mContext.resources.displayMetrics.density
        val params = AbsListView.LayoutParams(parent.width / 7 - dp.toInt(), (parent.height - dp.toInt() * mDateManager.weeks) / mDateManager.weeks)
        convertView.layoutParams = params

        //日付のみ表示させる
        val dateFormat = SimpleDateFormat("d", Locale.US)
        holder.dateText!!.text = dateFormat.format(dateArray[position])

        val format = SimpleDateFormat("yyyy/MM/dd", Locale.JAPAN)

        //当月以外のセルをグレーアウト
        when {
            searchPlan(format.format(dateArray[position])) !== "" -> convertView.setBackgroundColor(Color.rgb(210, 240, 200))
            mDateManager.isCurrentMonth(dateArray[position]) -> convertView.setBackgroundColor(Color.WHITE)
            else -> convertView.setBackgroundColor(Color.LTGRAY)
        }

        //日曜日を赤、土曜日を青に
        val colorId: Int = when (mDateManager.getDayOfWeek(dateArray[position])) {
            1 -> Color.RED
            7 -> Color.BLUE

            else -> Color.BLACK
        }
        holder.dateText!!.setTextColor(colorId)

        return convertView
    }

    override fun getItemId(position: Int): Long {
        return 0
    }

    override fun getItem(position: Int): Any? {
        return null
    }

    //翌月表示
    fun nextMonth() {
        mDateManager.nextMonth()
        dateArray = mDateManager.days
        this.notifyDataSetChanged()
    }

    //前月表示
    fun prevMonth() {
        mDateManager.prevMonth()
        dateArray = mDateManager.days
        this.notifyDataSetChanged()
    }

    private fun searchPlan(date: String): String {
        var plan: String

        dbAdapter = DBAdapter(this.mContext)
        dbAdapter.readDB()                         // DBの読み込み(読み込みの方)

        val column = "date"          //検索対象のカラム名
        val name = arrayOf(date)            //検索対象の文字

        // DBの検索データを取得 入力した文字列を参照してDBの品名から検索
        val c = dbAdapter.searchDB("remainder", null, column, name)

        c.moveToFirst()
        try {
            plan = c.getString(2)
            Log.d(date, plan)
        } catch (e: CursorIndexOutOfBoundsException) {

            Log.d(date, "null")
            plan = ""
        }

        c.close()
        dbAdapter.closeDB()        // DBを閉じる

        return plan
    }

    companion object {

        var dateArray: List<Date> = ArrayList()
    }

}
